Source/FieldSolver/WarpX_QED_K.H

```diff
@@ -0,0 +1,279 @@
+ * \param[in] tempEx Since the corrections to E at a given node are non-linear functions
+ *            of the values of E on the surronding nodes, temp arrays are used so that
+ *            modifications to one node do not influence the corrections to the surronding nodes
+ * \param[in] tempEy Since the corrections to E at a given node are non-linear functions
+ *            of the values of E on the surronding nodes, temp arrays are used so that modifications to
+ *            one node do not influence the corrections to the surronding nodes
+ * \param[in] tempEz Since the corrections to E at a given node are non-linear functions
+ *            of the values of E on the surronding nodes, temp arrays are used so that modifications to
+ *            one node do not influence the corrections to the surronding nodes
+ * \param[in] dx The x spatial step, used for calculating curls
+ * \param[in] dy The y spatial step, used for calculating curls
+ * \param[in] dz The z spatial step, used for calulating curls
+ * \param[in] dt The temporal step, used for the half push/correction to the E-fields at the end of the function
+ */
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void warpx_hybrid_QED_push (int j, int k, int l, Array4<Real> const& Ex, Array4<Real>
+                            const& Ey, Array4<Real> const& Ez, Array4<Real> const& Bx,
+                            Array4<Real> const& By, Array4<Real const> const& Bz,
+                            Array4<Real> const& tmpEx, Array4<Real> const& tmpEy,
+                            Array4<Real> const& tmpEz, Real dx, Real dy, Real dz, Real dt)
+{
+
+const amrex::Real c2 = PhysConst::c * PhysConst::c;
+const amrex::Real xi = WarpX::quantum_xi;
+const amrex::Real dxi = 1./dx;
+const amrex::Real dzi = 1./dz;
+
+#if (AMREX_SPACEDIM == 3)
+const amrex::Real dyi = 1./dy;
+
+    // Picking out points for stencil to be used in curl function of M
+
+    amrex::Real Mpx [3] = {0.,0.,0.};
+    amrex::Real Mnx [3] = {0.,0.,0.};
+    amrex::Real Mpy [3] = {0.,0.,0.};
+    amrex::Real Mny [3] = {0.,0.,0.};
+    amrex::Real Mpz [3] = {0.,0.,0.};
+    amrex::Real Mnz [3] = {0.,0.,0.};
+
+    calc_M(Mpx, tmpEx(j+1,k,l), tmpEy(j+1,k,l), tmpEz(j+1,k,l),
+                Bx(j+1,k,l), By(j+1,k,l), Bz(j+1,k,l), xi, c2);
+    calc_M(Mnx, tmpEx(j-1,k,l), tmpEy(j-1,k,l), tmpEz(j-1,k,l),
+                Bx(j-1,k,l), By(j-1,k,l), Bz(j-1,k,l), xi, c2);
+    calc_M(Mpy, tmpEx(j,k+1,l), tmpEy(j,k+1,l), tmpEz(j,k+1,l),
+                Bx(j,k+1,l), By(j,k+1,l), Bz(j,k+1,l), xi, c2);
+    calc_M(Mny, tmpEx(j,k-1,l), tmpEy(j,k-1,l), tmpEz(j,k-1,l),
+                Bx(j,k-1,l), By(j,k-1,l), Bz(j,k-1,l), xi, c2);
+    calc_M(Mpz, tmpEx(j,k,l+1), tmpEy(j,k,l+1), tmpEz(j,k,l+1),
+                Bx(j,k,l+1), By(j,k,l+1), Bz(j,k,l+1), xi, c2);
+    calc_M(Mnz, tmpEx(j,k,l-1), tmpEy(j,k,l-1), tmpEz(j,k,l-1),
+                Bx(j,k,l-1), By(j,k,l-1), Bz(j,k,l-1), xi, c2);
+
+    // Calculating necessary curls
```

Comment:
This kind of comments are useful. Could you add some more in the 3D section, as well as ins the 2D one?